### PR TITLE
feat(front): upgrade tiptap to v3

### PR DESCRIPTION
**Pull request by ElPicador**
## Tests

Conversation
* writing text ✅ 
* mentions  ✅ 
* Override Backspace behavior so it removes a single character from label ✅ 
* copy/pasting
	* code  ✅ 
	* markdown (does nothing as expected) ✅ 
	* mention 🚫 - adds a new line on chrome only (can't test on safari)
	* links of a datasource  ✅ 
* shift+enter ✅ 

Agent builder:
* writing text ✅ 
* mentions ✅ (doesn't do anything as expected)
* XML ✅ 
* copy/pasting
	* code ✅  
	* markdown ✅
	* mention ✅ (doesn't do anything as expected)
	* XML ❓ doesn't work in prod either
    * links of a datasource ✅ (doesn't do anything as expected)
* shift+enter  ✅ (doesn't do anything as expected)

### CODE REVIEW

File: extension/package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "analyze:chrome": "tsx run/build.ts --platform=chrome --analyze",
     "lint": "eslint .",
     "format": "prettier --write .",
-    "format:check": "prettier --check ."
+    "format:check": "prettier --check .",
+    "tsc": "tsc"
```
Comment (ElPicador) [inline extension/package.json]: added this as it was simpler for me to compile the extension

File: extension/ui/components/input_bar/editor/extensions/MentionStorageExtension.ts [hunk 2]
```diff
@@ -1,6 +1,14 @@
 import type { EditorSuggestions } from "@app/ui/components/input_bar/editor/suggestion";
 import { Extension } from "@tiptap/react";
 
+declare module "@tiptap/core" {
```
Comment (ElPicador) [inline extension/ui/components/input_bar/editor/extensions/MentionStorageExtension.ts]: to have the types correct

File: extension/ui/components/input_bar/editor/extensions/URLDetectionExtension.tsx [hunk 3]
```diff
@@ -53,6 +53,11 @@ export const URLDetectionExtension = Extension.create<URLFormatOptions>({
                   const nodeId = isUrlNodeCandidate
                     ? nodeCandidate.url
                     : nodeCandidate.node;
+
+                  if (!nodeId) {
```
Comment (ElPicador) [inline extension/ui/components/input_bar/editor/extensions/URLDetectionExtension.tsx]: the `nodeCandidate.url` can be null

File: extension/ui/components/input_bar/editor/useCustomEditor.tsx [hunk 4]
```diff
@@ -8,9 +8,9 @@ import { URLStorageExtension } from "@app/ui/components/input_bar/editor/extensi
 import { createMarkdownSerializer } from "@app/ui/components/input_bar/editor/markdownSerializer";
 import type { EditorSuggestions } from "@app/ui/components/input_bar/editor/suggestion";
 import type { SuggestionProps } from "@app/ui/components/input_bar/editor/useMentionDropdown";
-import { MentionPluginKey } from "@tiptap/extension-mention";
 import Paragraph from "@tiptap/extension-paragraph";
 import Placeholder from "@tiptap/extension-placeholder";
+import { PluginKey } from "@tiptap/pm/state";
```
Comment (ElPicador) [inline extension/ui/components/input_bar/editor/useCustomEditor.tsx]: it's part of the v2 upgrade I do in the extension at the same time (already applied in front)

File: front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx [hunk 5]
```diff
@@ -92,7 +92,7 @@ export function AgentBuilderInstructionsEditor({
         bold: false,
         italic: false,
         strike: false,
-        history: {
+        undoRedo: {
```
Comment (ElPicador) [inline front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx]: this was renamed see changelog

File: front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx [hunk 6]
```diff
@@ -141,6 +141,7 @@ export function AgentBuilderInstructionsEditor({
           debouncedUpdate(editor);
         }
       },
+      immediatelyRender: false,
```
Comment (ElPicador) [inline front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx]: to disable SSR

File: front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx [hunk 7]
```diff
@@ -186,10 +187,9 @@ export function AgentBuilderInstructionsEditor({
     if (currentContent !== field.value) {
       // Use setTimeout to ensure this runs after any diff mode changes
       setTimeout(() => {
-        editor.commands.setContent(
-          tipTapContentFromPlainText(field.value),
-          false
-        );
+        editor.commands.setContent(tipTapContentFromPlainText(field.value), {
+          emitUpdate: false,
```
Comment (ElPicador) [inline front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx]: signature of `setContent` changed, it's an object of options now

File: front/components/agent_builder/instructions/extensions/AgentInstructionDiffExtension.ts [hunk 8]
```diff
@@ -15,6 +16,18 @@ declare module "@tiptap/core" {
       exitDiff: () => ReturnType;
     };
   }
+
+  interface Storage {
```
Comment (ElPicador) [inline front/components/agent_builder/instructions/extensions/AgentInstructionDiffExtension.ts]: types types types

File: front/components/assistant/conversation/co_edition/CoEditionBubbleMenu.tsx [hunk 9]
```diff
@@ -21,8 +21,8 @@ export function CoEditionBubbleMenu({
   return (
     <BubbleMenu
       editor={editor}
-      tippyOptions={{
-        duration: 200,
+      options={{
```
Comment (ElPicador) [inline front/components/assistant/conversation/co_edition/CoEditionBubbleMenu.tsx]: it's not tippy internally, and duration doesn't exist anymore

File: front/components/assistant/conversation/co_edition/CoEditionContainer.tsx [hunk 10]
```diff
@@ -5,7 +5,7 @@ import {
   cn,
   XMarkIcon,
 } from "@dust-tt/sparkle";
-import Placeholder from "@tiptap/extension-placeholder";
+import { Placeholder } from "@tiptap/extensions";
```
Comment (ElPicador) [inline front/components/assistant/conversation/co_edition/CoEditionContainer.tsx]: see changelog, some imports change

File: front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx [hunk 11]
```diff
@@ -17,12 +17,6 @@ interface MentionExtensionOptions extends MentionOptions {
 }
 
 export const MentionExtension = Mention.extend<MentionExtensionOptions>({
-  addOptions() {
```
Comment (ElPicador) [inline front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx]: unneeded

File: front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx [hunk 12]
```diff
@@ -312,9 +313,18 @@ const useCustomEditor = ({
 
           const setLoading = (loading: boolean) => {
             if (loading) {
-              editor?.view.dom.classList.add("loading-text");
+              editor?.setOptions({
```
Comment (ElPicador) [inline front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx]: unsure about this change, but I'm not sure how to test

File: front/package.json [hunk 13]
```diff
@@ -55,13 +55,12 @@
     "@temporalio/worker": "^1.12.1",
     "@temporalio/workflow": "^1.12.1",
     "@textea/json-viewer": "^3.1.1",
-    "@tiptap/extension-character-count": "^2.26.1",
-    "@tiptap/extension-link": "^2.26.1",
```
Comment (ElPicador) [inline front/package.json:74]: link not needed it's in the starter kit now

File: front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx [hunk 14]
```diff
@@ -347,6 +347,7 @@ const useCustomEditor = ({
         return false;
       },
     },
+    immediatelyRender: false,
```
Comment (ykmsd) [inline front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx]: 📝 this is to not render on server side

File: front/components/assistant/conversation/input_bar/editor/extensions/ParagraphExtension.ts [hunk 15]
```diff
@@ -3,8 +3,6 @@ import { Paragraph } from "@tiptap/extension-paragraph";
 export const ParagraphExtension = Paragraph.extend({
   addKeyboardShortcuts() {
     return {
-      ...this.parent?.(),
```
Comment (ykmsd) [inline front/components/assistant/conversation/input_bar/editor/extensions/ParagraphExtension.ts:6]: probably super minor but we don't wanna keep the existing shortcuts? (which is [setParagraph()](https://tiptap.dev/docs/editor/extensions/nodes/paragraph#keyboard-shortcuts))

File: front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx (hunk 5 above)
Comment (ykmsd) [reply front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx]: weird name 🤨

File: front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx [hunk 16]
```diff
@@ -140,7 +138,6 @@ export const MentionExtension = Mention.extend<MentionExtensionOptions>({
   // the @-suggestion dropdown).
   addKeyboardShortcuts() {
     return {
-      ...(this.parent?.() ?? {}),
```
Comment (ykmsd) [inline front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx:143]: seems like this is okay to skip it since the default shortcut is only `Backspace` and we are overriding it?

File: front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx [hunk 17]
```diff
@@ -80,15 +79,14 @@ export const MentionExtension = Mention.extend<MentionExtensionOptions>({
...
-          this.editor.storage.MentionStorage.suggestions;
+        const suggestions = this.editor.storage.MentionStorage.suggestions;
 
         const results: PasteRuleMatch[] = suggestions.suggestions.flatMap(
           (suggestion) => {
             return [
               ...text.matchAll(
                 // Note: matching the @ that are found either at the start of a line or after a whitespace character.
-                // and that also are followed by a newline, a whitespace character or the end of the string.
+                // and that also is followed by a newline, a whitespace character or the end of the string.
```
Comment (ykmsd) [inline front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx]: super nitpick but actually isn't it supposed to be plural? the subject of this sentence should be `matching all the @mentions`?

**Review by ykmsd — APPROVED** [review]
I tested a lot and did my best to review the code and looks good, left a few comments here and there. But I'm a tiptap newbie so you might want to get a review from the specialist (aka flavien) 🙏 

also I fixed the type error to test it on front-edge, if you can take a look that would be great https://github.com/dust-tt/dust/pull/16306/commits/7df31709c06397476794c9b78c03772158c49b2d 🙇‍♀️ 

The only thing I spotted is that seems like any URLs will be clickable by default and it feels weird to me. Either we should apply clickable style or disable [openOnClick](https://tiptap.dev/docs/editor/extensions/marks/link#openonclick ) option, what do you think? 

https://github.com/user-attachments/assets/605d44e8-5c97-40a5-a107-fbdb81e5e6a0

File: front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx (hunk 5 above)
Comment (ElPicador) [reply front/components/agent_builder/instructions/AgentBuilderInstructionsEditor.tsx]: yup 🤷

File: front/components/assistant/conversation/input_bar/editor/extensions/ParagraphExtension.ts (hunk 15 above)
Comment (ElPicador) [reply front/components/assistant/conversation/input_bar/editor/extensions/ParagraphExtension.ts:6]: indeed! I was confused by the name `addXXX` but in fact it's overiddes everything

File: front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx (hunk 17 above)
Comment (ElPicador) [reply front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx]: indeed
